project-addons/custom_partner/partner.py

```diff
@@ -110,9 +173,25 @@ def _calculate_annual_invoiced(self):
                 move_ids = self.env['stock.move'].search([('picking_id', '=', picking.id)])
                 for move in move_ids:
```

Comment:
Podéis iterar sobre picking.move_lines y os evitáis una consulta y dos se aplicáis esto también arriba, es un método que va a necesitar su tiempo para ejecutarse y cada vez más, por lo que cuanto menos tenga que hacer mejor.